cloud/aws/elasticache/mode/discovery.pm

```diff
@@ -0,0 +1,114 @@
+#
+# Copyright 2022 Centreon (http://www.centreon.com/)
+#
+# Centreon is a full-fledged industry-strength solution that meets
+# the needs in IT infrastructure and application monitoring for
+# service performance.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package cloud::aws::elasticache::mode::discovery;
+
+use base qw(centreon::plugins::mode);
+
+use strict;
+use warnings;
+use JSON::XS;
+
+sub new {
+    my ($class, %options) = @_;
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options);
+    bless $self, $class;
+    
+    $options{options}->add_options(arguments => {
+        "prettify"  => { name => 'prettify' },
+    });
+    return $self;
+}
+
+sub check_options {
+    my ($self, %options) = @_;
+    $self->SUPER::init(%options);
+}
+
+sub run {
+    my ($self, %options) = @_;
+    my @disco_data;
+    my $disco_stats;
+
+    $disco_stats->{start_time} = time();
+   
+    my $instances = $options{custom}->discovery(
+        service => 'elasticache',
+        command => 'describe-cache-clusters'
+    );
+
+    foreach my $ecc_instance (@{$instances->{CacheClusters}}) {
+        next if (!defined($ecc_instance->{CacheClusterId}));
+            my %ecc;
+            $ecc{type}= "elasticache";
+            $ecc{id} = $ecc_instance->{CacheClusterId};
+            $ecc{engine} = $ecc_instance->{Engine};
+            $ecc{engine_version} = $ecc_instance->{EngineVersion};
+            $ecc{replication_group_log_delivery_enabled} = $ecc_instance->{ReplicationGroupLogDeliveryEnabled};
```

Comment:
i'm not sure that autodiscovery managed json boolean (true/false). maybe make it 0 or 1